components/Collections/examples/CollectionsInfoBarExample.m

```diff
@@ -0,0 +1,132 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "CollectionsInfoBarExample.h"
+
+static const NSInteger kItemCount = 10;
+static NSString *const kReusableIdentifierItem = @"itemCellIdentifier";
+
+@implementation CollectionsInfoBarExample {
+  NSArray *_accessoryTypes;
+  NSMutableArray *_content;
+  BOOL _enableEditing;
+  BOOL _allowsSwipeToDismiss;
+  BOOL _useCustomInfoBar;
+}
+
++ (NSArray *)catalogBreadcrumbs {
+  return @[ @"Collections", @"Collections InfoBar Example" ];
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  // Register cell class.
+  [self.collectionView registerClass:[MDCCollectionViewTextCell class]
+          forCellWithReuseIdentifier:kReusableIdentifierItem];
+
+  // Populate content.
+  _content = [NSMutableArray array];
+  [_content addObject:@[ @"Enable Editing", @"Enable Swipe-to-Dismiss"]];
+  NSMutableArray *items = [NSMutableArray array];
+  for (NSInteger i = 0; i < kItemCount; i++) {
+    [items addObject:[NSString stringWithFormat:@"Item-%zd", i]];
+  }
+  [_content addObject:items];
+
+  // Customize collection view settings.
+  _enableEditing = NO;
+  _allowsSwipeToDismiss = YES;
+  self.styler.cellStyle = MDCCollectionViewCellStyleCard;
+}
+
+#pragma mark - <UICollectionViewDataSource>
+
+- (NSInteger)numberOfSectionsInCollectionView:(UICollectionView *)collectionView {
+  return [_content count];
+}
+
+- (NSInteger)collectionView:(UICollectionView *)collectionView
+     numberOfItemsInSection:(NSInteger)section {
+  return [_content[section] count];
+}
+
+- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView
+                  cellForItemAtIndexPath:(NSIndexPath *)indexPath {
+  MDCCollectionViewTextCell *cell =
+      [collectionView dequeueReusableCellWithReuseIdentifier:kReusableIdentifierItem
+                                                forIndexPath:indexPath];
+  cell.textLabel.text = _content[indexPath.section][indexPath.item];
+
+  // Add switched as accessory views.
+  if (indexPath.section == 0) {
+    UISwitch *switchControl = [[UISwitch alloc] initWithFrame:CGRectZero];
```

Comment:
This pattern seems unfamiliar.
I wouldn't expect to allocate and add views to a cell here.  I would typically do so in the cell's init.
SwitchCell would inherit from MDCCollectionViewTextCell.  TextCell and SwitchCell would have different identifiers which they use to dequeue.